include/cuda/memory_resource

```diff
@@ -0,0 +1,937 @@
+ * which can be either one of the base resource classes or a concrete resource type.
+ *
+ * \tparam _ResourcePointer a pointer-like object to the underlying memory resource
+ * \tparam _Properies properties of a memory resource required by resource view
+ */
+template <typename _ResourcePointer, typename... _Properties>
+class basic_resource_view {
+public:
+    static_assert(
+        _CUDA_VSTD::is_base_of<detail::memory_resource_base, typename ::std::pointer_traits<_ResourcePointer>::element_type>::value ||
+        _CUDA_VSTD::is_base_of<detail::stream_ordered_memory_resource_base, typename ::std::pointer_traits<_ResourcePointer>::element_type>::value,
+        "ResourcePointer must be a pointer to a memory_resource_base, stream_ordered_memory_resource_base or a derived class");
+
+    basic_resource_view() = default;
+
+    basic_resource_view(int) = delete;
+
+    basic_resource_view(std::nullptr_t) {}
+
+    /*!
+     * \brief Constructs a resource view from a compatible memory resource pointer.
+     *
+     * The memory resource is considered compatible if a pointer to it can be converted to
+     * `_ResourcePointer` and the resource type has the required properties listed
+     * in `_Properties`.
+     *
+     * \tparam _Resource Type of a mmeory resource object.
+     * \param __p pointer to a memory resource object.
+     */
+    template <typename _Resource, typename = _CUDA_VSTD::enable_if_t<
+        detail::is_resource_pointer_convertible<_Resource*, _ResourcePointer>::value &&
+        _CUDA_VSTD::conjunction<has_property<_Resource, _Properties>...>::value
+    >>
+    basic_resource_view(_Resource *__p) : __pointer(__p) {}
+
+    /*!
+     * \brief Constructs a resource view by copying the resource pointer from a compatible resource view.
+     *
+     * A resource view is considered compatible if it defines all properties required by this
+     * view in `_Properties`.
+     *
+     * \tparam _OtherPointer The resource pointer type of the source resource view
+     * \tparam _OtherProperties The properties defined byt the source resource view
+     */
+    template <typename _OtherPointer, typename... _OtherProperties, typename = _CUDA_VSTD::enable_if_t<
+        is_view_convertible<basic_resource_view<_OtherPointer, _OtherProperties...>, basic_resource_view>::value
+    >>
+    basic_resource_view(basic_resource_view<_OtherPointer, _OtherProperties...> v) : __pointer(v.__pointer) {}
+
+    /*!
+     * \brief Exposes the interface of the underlying memory resource.
+     *
+     * \note This method should not be used to obtain the pointer to the memory resource.
+     */
+    _ResourcePointer operator->() const noexcept { return __pointer; }
+
+    template <typename _Ptr2, typename... _Props2>
+    bool operator==(const cuda::basic_resource_view<_Ptr2, _Props2...> &__v2) const noexcept {
+      using __view1_t = basic_resource_view;
+      using __view2_t = basic_resource_view<_Ptr2, _Props2...>;
```

Comment:
These two typedefs are unused so they cause a warning with g++9. We treat warnings as errors in RMM so this breaks the build. Suggest removing these. 
```suggestion
```